build.gradle

```diff
@@ -389,6 +390,64 @@ task checkVersion {
 	}
 }
 
+// Authentication is provided by GRGIT_PASS and GRGIT_USER enviro. variables.
+// GRGIT_PASS can be authentication token
+// See https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token#using-a-token-on-the-command-line
+def versionMapRepo = "https://github.com/dexman545/fabric-api-version-map.git" //TODO use official repo
+gitPublish {
+	repoUri = versionMapRepo
+	branch = "main"
+	commitMessage = "Automated addition of new api version from FabricMC repo"
+
+	preserve {
+		include "README.md"
+		include "LICENSE"
+		include "apiVersions.properties"
+		include ".gitignore"
+	}
```

Comment:
I personally suggest moving these stuff to github actions than using another gradle plugin which is useless in most of the use cases.

Reply:
Can actions read and write a properties file in a sane way?
I can replace the plugin and use grgit or an `Exec` task if the adding of a new plugin is undesired.

Reply:
Imo a exec would be good enough. It would avoid problems like upgrading gradle and having this plugin break because it doesn't support new gradle, and people who just want to develop api would not need to grab this gradle plugin.